src/cascadia/TerminalSettingsEditor/GlobalAppearance.cpp

```diff
@@ -28,4 +35,139 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+
+        // [1]:
+        {
+            // ManifestLanguages contains languages the app ships with.
+            //
+            // Languages is a computed list that merges the ManifestLanguages with the
+            // user's ranked list of preferred languages taken from the system settings.
+            // As is tradition the API documentation is incomplete though, as it can also
+            // contain regional language variants. If our app supports en-US, but the user
+            // has en-GB or en-DE in their system's preferred language list, Languages will
+            // contain those as well, as they're variants from a supported language. We should
+            // allow a user to select those, as regional formattings can vary significantly.
+            const std::array tagSources{
+                winrt::Windows::Globalization::ApplicationLanguages::ManifestLanguages(),
+                winrt::Windows::Globalization::ApplicationLanguages::Languages()
+            };
+
+            // tags will hold all the flattened results from tagSources.
+            // We resize() the vector to the proper size in order to efficiently GetMany() all items.
+            tags.resize(std::accumulate(
+                tagSources.begin(),
+                tagSources.end(),
+                // tags[0] will be "und" - the "Use system language" item
+                // tags[1..n] will contain tags from tagSources.
+                // --> totalTags is offset by 1
+                1,
+                [](uint32_t sum, const auto& v) -> uint32_t {
+                    return sum + v.Size();
+                }));
+
+            // As per the function definition, the first item
+            // is always "Use system language" ("und").
+            auto data = tags.data();
+            *data++ = systemLanguageTag;
+
+            // Finally GetMany() all the tags from tagSources.
+            for (const auto& v : tagSources)
+            {
+                const auto size = v.Size();
+                v.GetMany(0, winrt::array_view(data, size));
+                data += size;
+            }
+        }
+
+        // NOTE: The size of tags is always >0, due to tags[0] being hardcoded to "und".
+        const auto tagsBegin = ++tags.begin();
+        const auto tagsEnd = tags.end();
+
+        // [2]:
+        std::sort(tagsBegin, tagsEnd);
+
+        // I'd love for both, std::unique and std::remove_if, to occur in a single loop,
+        // but the code turned out to be complex and even less maintainable, so I gave up.
+        {
+            // [3] part 1:
+            auto it = std::unique(tagsBegin, tagsEnd);
+
+            // The qps- languages are useful for testing ("pseudo-localization").
+            // --> Leave them in if debug features are enabled.
+            if (!_State.Globals().DebugFeaturesEnabled())
+            {
+                // [4] part 1:
+                it = std::remove_if(it, tagsEnd, [](const winrt::hstring& tag) mutable -> bool {

```

Comment:
@DHowett FYI I just realized this is a programming error and should be `std::remove_if(tagsBegin, it, ...)` instead. `std::unique` returns an iterator for the new end of the range, which I have to use instead of `tagsEnd` here.